### PR TITLE
Education benefits navigation related updates

### LINKED ISSUE #2871: Navigation updates for edu-benefits
- The wireframes for mobile have a step count at the top of each section.
- We shouldn't show page name if only one name in chapter
- Use route list to get panel order and grouping instead of Redux store
- Stop text from wrapping underneath checkboxes

### PR DESCRIPTION
**Pull request by jbalboni**
Closes #2871.

The changes are listed in that issue. I moved the chapter/panel data out of the Redux store and into props on the routes list. I think this makes more sense and gives us a single place to look for any information related to the order and grouping of the pages of the form.

I didn't rename sections to pages, but we should create an issue for that.

### CODE REVIEW

**Comment by jbalboni** [comment]
@alexose Want to take this one, since you reviewed the earlier nav PR I did?

**Comment by jbalboni** [comment]
screen shot 2016-08-30 at 9 59 46 am

File: src/js/edu-benefits/routes.jsx [hunk 1]
```diff
@@ -13,68 +15,83 @@ const routes = [
...
-      path="/school-selection/school-information"/>,
+      path="/school-selection/school-information"
+      chapter={chapters.schoolSelection}/>,
   // Review and Submit route.
   <Route
       component={PlaceholderSection}
       key="/review-and-submit"
-      path="/review-and-submit"/>,
+      path="/review-and-submit"
+      chapter={chapters.review}/>,
```
Comment (alexose) [inline src/js/edu-benefits/routes.jsx]: Part of me wants to write this in a DRY-er way.  But, I think this reads well, and additional magic wouldn't improve that.

Speaking of readability, I think a comment describing what Chapters are could be useful.

File: src/js/edu-benefits/components/Nav.jsx [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
-  return panel.sections.slice(-1)[0].path;
+function lastSection(chapter) {
+  return chapter.sections.slice(-1)[0].path;
 }
 
-function determinePanelStyles(path, sectionState, formPanel, currentUrl) {
+function determineChapterStyles(sectionState, formChapter, currentUrl) {
   let classes = '';
-  if (currentUrl.startsWith(path)) {
+  if (formChapter.sections.some(section => section.path === currentUrl)) {
```
Comment (alexose) [inline src/js/edu-benefits/components/Nav.jsx]: Array.some()!  Hooray ES2015.

**Comment by alexose** [comment]
This is looking great so far.  One thing I might do is to store the results of `groupPagesIntoChapters` somewhere (perhaps export it from `routes.jsx`?) so that we don't have to re-run it in each container.

**Comment by jbalboni** [comment]
@alexose Pushed some updates that address your feedback. Some other things had to change around because NavButtons moved to the main app component.

**Comment by alexose** [comment]
👍

**Pull request merged**